File: video-streaming-frontend/src/app/features/video/components/comments/models/comment.dto.ts
import { UserDto } from "../../../../../shared/models/user.dto";
import { UserLikeStatus } from "../../../../../shared/models/videoLikeStatus.dto";



export interface CommentDto {
	id: number;
	text: string;
	user: UserDto;
	likesCount: number;
	dislikesCount: number;
	userLikeStatus?: UserLikeStatus;
	createdDate: Date;
}

File: video-streaming-backend/src/main/java/com/fourbit/sachintha/video_streaming_platform/dto/CommentDto.java
package com.fourbit.sachintha.video_streaming_platform.dto;

import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CommentDto {
	private Long id;
	private String text;
	private UserDto user;
	private Long likesCount = Long.valueOf(0);
	private Long dislikesCount = Long.valueOf(0);
	private UserLikeStatus userLikeStatus;
	private LocalDateTime createdDate;
}
